src/utils.cpp

```diff
@@ -148,6 +149,23 @@ void USDTHelper::read_probes_for_path(const std::string &path)
   provider_cache_loaded = true;
 }
 
+std::string get_pid_exe(pid_t pid)
+{
+  char proc_path[512];
+  char exe_path[4096];
+  int res;
+
+  sprintf(proc_path, "/proc/%d/exe", pid);
+  res = readlink(proc_path, exe_path, sizeof(exe_path));
+  if (res == -1)
+    return "";
+  if (res >= static_cast<int>(sizeof(exe_path)))
+    res = sizeof(exe_path) - 1;
```

Comment:
I think we should return some kind of error if the path is too long for our buffer here. Otherwise we could get a link to the wrong exe